File: API/src/main/java/org/example/data/dto/AnalyseRequest.java
package org.example.data.dto;

import jakarta.validation.constraints.Positive;
import jakarta.validation.constraints.Size;
import lombok.*;

import java.util.List;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class AnalyseRequest {
    @Size(min = 1, message = "At least one candle is required")
    private List<HistoricCandleDTO> candles;

    @Positive(message = "Risk free rate must be positive")
    private Double riskFree;

    @Positive(message = "Mean benchmark must be positive")
    private Double meanBenchmark;
}

File: Analyse/src/main/kotlin/com/example/dto/AnalyseRequest.kt
package com.example.dto

import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class AnalyseRequest(
    @SerialName("candles")
    val candles: List<Candle>,

    @SerialName("risk_free")
    val riskFree: Double,

    @SerialName("mean_benchmark")
    val meanBenchmark: Double,
)
